index.js

```diff
@@ -2,21 +2,23 @@
 const path = require('path');
 const resolveCwd = require('resolve-cwd');
 const pkgDir = require('pkg-dir');
+const { fileURLToPath } = require('url');
 
 module.exports = filename => {
-	const globalDir = pkgDir.sync(path.dirname(filename));
-	const relativePath = path.relative(globalDir, filename);
-	const pkg = require(path.join(globalDir, 'package.json'));
-	const localFile = resolveCwd.silent(path.join(pkg.name, relativePath));
-	const localNodeModules = path.join(process.cwd(), 'node_modules');
+  const normalizedFilename = filename.startsWith('file://') ? fileURLToPath(filename) : filename
+  const globalDir = pkgDir.sync(path.dirname(normalizedFilename));
+  const relativePath = path.relative(globalDir, normalizedFilename);
+  const pkg = require(path.join(globalDir, 'package.json'));
+  const localFile = resolveCwd.silent(path.join(pkg.name, relativePath));
+  const localNodeModules = path.join(process.cwd(), 'node_modules');
 
-	const filenameInLocalNodeModules = !path.relative(localNodeModules, filename).startsWith('..') &&
-		// On Windows, if `localNodeModules` and `filename` are on different partitions, `path.relative()` returns the value of `filename`, resulting in `filenameInLocalNodeModules` incorrectly becoming `true`.
-		path.parse(localNodeModules).root === path.parse(filename).root;
+  const filenameInLocalNodeModules = !path.relative(localNodeModules, normalizedFilename).startsWith('..') &&
```

Comment:
Please don't make unrelated changes (the indentation).

Reply:
sorry, my fault. now, my modifications follow the lint information